lib/instrumentation/modules/@aws-sdk/client-s3.js

```diff
@@ -178,33 +179,26 @@ function s3MiddlewareFactory (client, agent) {
       options: { step: 'initialize', priority: 'high', name: 'elasticAPMSpan' }
     },
     {
-      middleware: (next, context) => async (args) => {
```

Comment:
Note for reviewer: this logic is shared among other clients (S3 for now) so its moved to a specific module. I've kept options controlled in the client so we can stack more specific logic if needed